readthedocs/builds/views.py

```diff
@@ -54,14 +54,32 @@ def post(self, request, project_slug):
             return HttpResponseForbidden()
 
         version_slug = request.POST.get('version_slug')
+        commit = request.POST.get('commit')
+        build_pk = request.POST.get('build_pk')
+
         version = get_object_or_404(
             self._get_versions(project),
             slug=version_slug,
         )
 
+        # Set either the build or None
+        build = Build.objects.filter(pk=build_pk).first()
```

Comment:
If a `build_pk` from a different project is passed through the form maybe we should check it before we start any tasks.

```suggestion
        build = version.builds.filter(pk=build_pk).first()
```